kafka/header.go

```diff
@@ -0,0 +1,67 @@
+package kafka
+
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"fmt"
+	"strconv"
+)
+
+/*
+#include <string.h>
+#include <librdkafka/rdkafka.h>
+#include "glue_rdkafka.h"
+*/
+import "C"
+
+// Header represents a single Kafka message header.
+//
+// Message headers are made up of a list of Header elements, retaining their original insert
+// order and allowing for duplicate Keys.
+//
+// Key is a human readable string identifying the header.
+// Value is the key's binary value, Kafka does not put any restrictions on the format of
+// of the Value but it should be made relatively compact.
+// The value may be a byte array, empty, or nil.
+//
+// NOTE: Message headers are not available on producer delivery report messages.
+type Header struct {
```

Comment:
it seems this is a current librdkafa limitation, right?

in the C# case, I have code in to retrieve the headers for delivery reports, but always get back an empty list.

Reply:
Yep, I didn't think it added much value to retain headers to the delivery report.
I think we're fine documenting this fact until a proper use-case emerges, what do you think?

Reply:
I would have probably erred the other way since I usually prefer to optimize on having default behavior that doesn't surprise, but I don't think this is a big deal and I may be mis-prioritizing arguments the other way.

Reply:
Well, once functionality is added we can't remove or alter it, so I'd rather wait for a use-case.

Reply:
Oh, to be clear, headers are available in the delivery report callback in librdkafka, it is just in the Go bindings that I've skipped them since there is a performance penalty to each C call.
We'll add support for this when a use-case emerges from the community.

Reply:
interesting - i thought I saw this wasn't working. Yeah, I think excluding this is very justifiable. In C# I expect I'll consider the degree of overhead annoying but acceptable, and will likely include it (but will benchmark).